src/Bicep.Core/SemanticModel/Compilation.cs

```diff
@@ -14,41 +15,46 @@ namespace Bicep.Core.SemanticModel
     public class Compilation
```

Comment:
Have you looked how Roslyn modeled similar things? I wonder if there are any approaches we could borrow from there...

Reply:
Yeah, I've followed their design roughly here.

Reply:
I didn't look at that aspect too closely before, so that makes sense.